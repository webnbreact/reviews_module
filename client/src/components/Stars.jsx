import React from 'react';
import { FlexContainer_Row_SpaceBetween, StarStyle } from './Styles.js'; 

const Stars = (props) => {

  const star = <i className="fas fa-star" />;
  const halfStar = <i className="fas fa-star-half" />;

  const generateArrayOfStars = (average) => {
    let arrOfStars = [];
    for(var i = 1; i < average; i++) {
      arrOfStars.push(star);
    }
    if(average % 1 >= 0.75) {
      arrOfStars.push(star)
    }
    if(average % 1 > 0.25 && average % 1 < 0.75) {
      arrOfStars.push(halfStar);
    }
    return arrOfStars;
  }

  const starsAccuracy = generateArrayOfStars(props.stars.accuracy);
  const starsCommunication = generateArrayOfStars(props.stars.communication);
  const starsCleanliness = generateArrayOfStars(props.stars.cleanliness);
  const starsLocation = generateArrayOfStars(props.stars.location);
  const starsCheckin = generateArrayOfStars(props.stars.check_in);
  const starsValue = generateArrayOfStars(props.stars.value);

  return ( 
    <FlexContainer_Row_SpaceBetween>
      <div><span>Accuracy</span><br></br><span>Communication</span><br></br><span>Cleanliness</span></div>
      <div>
        <span>
        {starsAccuracy.map((star, index) => {
          return (
            <StarStyle key={index}>
              {star}
            </StarStyle> 
            )
          }
        )}  
        </span>   
        <br></br>
        <span>
        {starsCommunication.map((star, index) => {
          return (
            <StarStyle key={index}>
              {star}
            </StarStyle> 
            )
          }
        )}  
        </span>
        <br></br>
        <span>
        {starsCleanliness.map((star, index) => {
          return (
            <StarStyle key={index}>
              {star}
            </StarStyle> 
            )
          }
        )}      
        </span>
      </div>
      <div><span>Location</span><br></br><span>Check-in</span><br></br><span>Value</span></div>
      <div>
        <span>
        {starsLocation.map((star, index) => {
          return (
            <StarStyle key={index}>
              {star}
            </StarStyle> 
            )
          }
        )}  
        </span>   
        <br></br>
        <span>
        {starsCheckin.map((star, index) => {
          return (
            <StarStyle key={index}>
              {star}
            </StarStyle> 
            )
          }
        )}  
        </span>
        <br></br>
        <span>
        {starsValue.map((star, index) => {
          return (
            <StarStyle key={index}>
              {star}
            </StarStyle> 
            )
          }
        )}      
        </span>
      </div>
    </FlexContainer_Row_SpaceBetween>
  )

}

export default Stars;
