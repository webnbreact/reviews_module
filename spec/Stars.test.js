import React from 'react';
import { shallow } from 'enzyme';
import Stars from '../client/src/components/Stars.jsx';
import { stars } from './testDummyData.js' ;

describe('Stars', () => {
  it('should be defined', () => {
    expect(Stars).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(
      <Stars name="Star test" stars={stars} />
    );
    expect(tree).toMatchSnapshot();
  })
})
