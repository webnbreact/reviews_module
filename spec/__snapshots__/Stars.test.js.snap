// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Stars
    name="Star test"
    stars={
      Array [
        Object {
          "accuracy": 2.3667,
          "check_in": 2.0333,
          "cleanliness": 1.9333,
          "communication": 1.6333,
          "location": 1.9333,
          "value": 1.7,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <span>
            <b>
              Accuracy
            </b>
          </span>
          <span>
            <b>
              Location
            </b>
          </span>
        </div>,
        <div>
          <span>
            <b>
              Communication
            </b>
          </span>
          <span>
            <b>
              Check-in
            </b>
          </span>
        </div>,
        <div>
          <span>
            <b>
              Cleanliness
            </b>
          </span>
          <span>
            <b>
              Value
            </b>
          </span>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              <b>
                Accuracy
              </b>
            </span>,
            <span>
              <b>
                Location
              </b>
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Accuracy
                </b>,
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Accuracy",
                },
                "ref": null,
                "rendered": "Accuracy",
                "type": "b",
              },
              undefined,
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Location
                </b>,
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Location",
                },
                "ref": null,
                "rendered": "Location",
                "type": "b",
              },
              undefined,
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              <b>
                Communication
              </b>
            </span>,
            <span>
              <b>
                Check-in
              </b>
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Communication
                </b>,
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Communication",
                },
                "ref": null,
                "rendered": "Communication",
                "type": "b",
              },
              undefined,
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Check-in
                </b>,
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Check-in",
                },
                "ref": null,
                "rendered": "Check-in",
                "type": "b",
              },
              undefined,
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              <b>
                Cleanliness
              </b>
            </span>,
            <span>
              <b>
                Value
              </b>
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Cleanliness
                </b>,
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cleanliness",
                },
                "ref": null,
                "rendered": "Cleanliness",
                "type": "b",
              },
              undefined,
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Value
                </b>,
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Value",
                },
                "ref": null,
                "rendered": "Value",
                "type": "b",
              },
              undefined,
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <span>
              <b>
                Accuracy
              </b>
            </span>
            <span>
              <b>
                Location
              </b>
            </span>
          </div>,
          <div>
            <span>
              <b>
                Communication
              </b>
            </span>
            <span>
              <b>
                Check-in
              </b>
            </span>
          </div>,
          <div>
            <span>
              <b>
                Cleanliness
              </b>
            </span>
            <span>
              <b>
                Value
              </b>
            </span>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                <b>
                  Accuracy
                </b>
              </span>,
              <span>
                <b>
                  Location
                </b>
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Accuracy
                  </b>,
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Accuracy",
                  },
                  "ref": null,
                  "rendered": "Accuracy",
                  "type": "b",
                },
                undefined,
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Location
                  </b>,
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Location",
                  },
                  "ref": null,
                  "rendered": "Location",
                  "type": "b",
                },
                undefined,
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                <b>
                  Communication
                </b>
              </span>,
              <span>
                <b>
                  Check-in
                </b>
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Communication
                  </b>,
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Communication",
                  },
                  "ref": null,
                  "rendered": "Communication",
                  "type": "b",
                },
                undefined,
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Check-in
                  </b>,
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Check-in",
                  },
                  "ref": null,
                  "rendered": "Check-in",
                  "type": "b",
                },
                undefined,
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                <b>
                  Cleanliness
                </b>
              </span>,
              <span>
                <b>
                  Value
                </b>
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Cleanliness
                  </b>,
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Cleanliness",
                  },
                  "ref": null,
                  "rendered": "Cleanliness",
                  "type": "b",
                },
                undefined,
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Value
                  </b>,
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Value",
                  },
                  "ref": null,
                  "rendered": "Value",
                  "type": "b",
                },
                undefined,
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
